packages/daemon/src/directory.js

```diff
@@ -97,6 +97,30 @@ export const makeDirectoryMaker = ({
       return formatLocator(id, formulaType);
     };
 
+    /** @type {import('./types.js').EndoDirectory['reverseLocate']} */
+    const reverseLocate = async locator => {
+      const id = idFromLocator(locator);
+      return petStore.reverseIdentify(id);
+    };
+
+    /** @type {import('./types.js').EndoDirectory['followLocatorNameChanges']} */
+    const followLocatorNameChanges = async function* followLocatorNameChanges(
+      locator,
+    ) {
+      const id = idFromLocator(locator);
+      for await (const idDiff of petStore.followIdNameChanges(id)) {
```

Comment:
I will digest this more easily if we only use “change” over synonyms like “diff” and “patch”. I take synonyms mean a deliberate difference and distinction.